src/dune_engine/scheduler.mli

```diff
@@ -9,11 +9,27 @@ module Config : sig
       | Short  (** One line per command *)
       | Verbose  (** Display all commands fully *)
 
-    type t =
+    type t = private
```

Comment:
We make this record private so that we can still use the projections, but we don't want people creating their own. They should go through the ones we have in the API. That way we can add extra data here and not worry about having to update other occurrences correctly.